pipeline/resources/postgis_geopandas_io_manager.py

```diff
@@ -71,14 +71,21 @@ def handle_output(self, context: OutputContext, obj: pandas.DataFrame):
         if isinstance(obj, pandas.DataFrame):
             with connect_postgresql(config=self._config) as con:
                 self._create_schema_if_not_exists(schema, con)
-                # just recreate table with empty frame (obj[:0]) and load later via copy_from
-                obj[:0].to_sql(
-                    con=con,
-                    name=table,
-                    index=True,
-                    schema=schema,
-                    if_exists='replace',
-                )
+                table_exists = self._has_table(con, schema, table)
```

Comment:
Does this run within a transaction? Or do we have other guarantees that, at any point in time, only one "instance" of this code will run?

Reply:
Code is run with a unique connection/transaction, yes.